.github/workflows/run-mypy-on-modified-cdk-files.yml

```diff
@@ -22,7 +23,7 @@ jobs:
         id: changed-py-files
         uses: tj-actions/changed-files@v23
         with:
-          files: "airbyte-cdk/python/airbyte_cdk/**/*.py"
+          files: "**/*.py"
       - name: Run if any of the listed files above is changed
         if: steps.changed-py-files.outputs.any_changed == 'true'
         run: mypy ${{ steps.changed-py-files.outputs.all_changed_files }} --config-file airbyte-cdk/python/mypy.ini --install-types --non-interactive
```

Comment:
@alafanechere that's the PR where @girarda prorotyped running `mypy` against connectors. 

I like how it's elegant and would hypothetically work. The trade-off is:
- Since this is not a dev dependency of a specific connector via `pyproject.toml`, it's not how the industry usually does this.
- We loose flexibility on config — using airbyte_cdk's config.
- I don't know if mypy won't freak out and will understand imports and module visibility correctly.

Reply:
> We loose flexibility on config — using airbyte_cdk's config.
I'm willing to call this a plus since it standardizes the mypy checks across all our connectors